src/hyperfocus/console/commands/add.py

```diff
@@ -2,25 +2,58 @@
 
 import click
 
+from hyperfocus.console.core.parameters import NotRequired
+from hyperfocus.console.core.parameters import NotRequiredIf
 from hyperfocus.services.session import get_current_session
 from hyperfocus.termui import formatter
 from hyperfocus.termui import printer
 from hyperfocus.termui.components import SuccessNotification
 
 
 @click.command(help="Add task to current working day")
-@click.argument("title", metavar="<title>", type=click.STRING)
+@click.argument(
+    "title",
+    cls=NotRequiredIf,
+    not_required_if=["bulk"],
+    metavar="<title>",
+    type=click.STRING,
+)
+@click.option(
+    "-d",
+    "--details",
+    "details",
+    cls=NotRequired,
+    not_required=["bulk"],
```

Comment:
Shouldn't it be `not_required_if` ?

Reply:
It's two different classes, see here: [parameters.py](https://github.com/u8slvn/hyperfocus/blob/main/src/hyperfocus/console/core/parameters.py)

The first one `NotRequiredIf` is a `click.Argument` class, it allows marking a command arg as "not required if" the following parameters are present. The second one `NotRequired` is a `click.Option` class, it raises an error if one of the given parameters as unnecessary if the option is present.

I guess it's time for me to add some docstrings!

Reply:
If I understood you well, it means that the `title` argument is not required when the `bulk` option is setup. But what if both `title` and `bulk` are presents ? Shouldn't it raise an error ?

I mean, the `bulk` option seems to be incompatible with both `details` and `title` since those two last arguments refer to a single task, while `bulk` introduces several tasks.

If indeed it is incompatible, either one should raise an error when setup along with `bulk`, what do you think ?

Reply:
This is already how it works.

Reply:
Ok fine.